test/amm/AmmPoolsExchangeRateAndSoap.t.sol

```diff
@@ -0,0 +1,491 @@
+
+        vm.prank(_liquidityProvider);
+        _iporProtocol.ammPoolsService.provideLiquidityDai(_liquidityProvider, TestConstants.USD_60_000_18DEC);
+
+        _iporProtocol.ammOpenSwapService.openSwapPayFixed28daysDai(
+            _userTwo,
+            26000 * TestConstants.D18,
+            9 * TestConstants.D17,
+            TestConstants.LEVERAGE_18DEC
+        );
+
+        vm.warp(block.timestamp + TestConstants.PERIOD_25_DAYS_IN_SECONDS);
+        (, , int256 soap) = _iporProtocol.ammSwapsLens.getSOAP(address(_iporProtocol.asset));
+
+        IporTypes.AmmBalancesMemory memory balance = _iporProtocol.ammPoolsLens.getBalance(
+            address(_iporProtocol.asset)
+        );
+
+        // when
+        uint256 actualExchangeRate = _iporProtocol.ammPoolsLens.getExchangeRate(address(_iporProtocol.asset));
+
+        // then
+        assertLt(soap, TestConstants.ZERO_INT);
+        assertLt(soap * -1, int256(balance.liquidityPool));
+        assertEq(actualExchangeRate, 1003704116338992634);
+    }
+
+    function testShouldCalculateExchangeRateReceiveFixedWhenSOAPChangedAndSOAPIsLowerThanZeroAndSOAPAbsoluteValueIsLowerThanLiquidityPoolBalance()
+        public
+    {
+        // given
+        //        _cfg.spread28DaysTestCase = BuilderUtils.Spread28DaysTestCase.CASE1;
+        _iporProtocol = _iporProtocolFactory.getDaiInstance(_cfg);
+
+        vm.prank(_userOne);
+        _iporProtocol.iporOracle.updateIndex(address(_iporProtocol.asset), TestConstants.PERCENTAGE_3_18DEC);
+
+        vm.prank(address(_liquidityProvider));
+        _iporProtocol.ammPoolsService.provideLiquidityDai(_liquidityProvider, TestConstants.USD_60_000_18DEC);
+
+        _iporProtocol.ammOpenSwapService.openSwapReceiveFixed28daysDai(
+            _userTwo,
+            27000 * TestConstants.D18,
+            1 * TestConstants.D16,
+            TestConstants.LEVERAGE_18DEC
+        );
+
+        vm.prank(_userOne);
+        _iporProtocol.iporOracle.updateIndex(address(_iporProtocol.asset), TestConstants.PERCENTAGE_5_18DEC);
+
+        vm.warp(block.timestamp + TestConstants.PERIOD_25_DAYS_IN_SECONDS);
+        (, , int256 soap) = _iporProtocol.ammSwapsLens.getSOAP(address(_iporProtocol.asset));
+
+        IporTypes.AmmBalancesMemory memory balance = _iporProtocol.ammPoolsLens.getBalance(
+            address(_iporProtocol.asset)
+        );
+
+        // when
+        uint256 actualExchangeRate = _iporProtocol.ammPoolsLens.getExchangeRate(address(_iporProtocol.asset));
+
+        // then
+        assertLt(soap, TestConstants.ZERO_INT);
+        assertLt(soap * -1, int256(balance.liquidityPool));
+        assertEq(actualExchangeRate, 1009369072035123743);
+    }
+
+    function testShouldCalculateExchangeRatePayFixedWhenSOAPChangedAndSOAPIsGreaterThanZeroAndSOAPAbsoluteValueIsLowerThanLiquidityPoolBalance()
+        public
+    {
+        // given
+        _cfg.spread28DaysTestCase = BuilderUtils.Spread28DaysTestCase.CASE0;
```

Comment:
In old test spread for pay fixed was 4% so here you should use Spread28DaysTestCase.CASE1

_cfg.spread28DaysTestCase = BuilderUtils.Spread28DaysTestCase.CASE1;

Reply:
done